setup.py

```diff
@@ -34,28 +34,33 @@ def read_response(cmd):
     return subprocess.check_output(cmd).decode("utf").strip()
 
 
-ext_options = {
-    "include_dirs": [],
-    "library_dirs": [],
-    "libraries": [],
-    "extra_link_args": [],
-}
-ext_modules = []
-
-
-# setuptools clean does not cleanup Cython artifacts
-if "clean" in sys.argv:
-    if os.path.exists("build"):
-        shutil.rmtree("build")
-
-    root = Path(".")
-    for ext in ["*.so", "*.pyc", "*.c", "*.cpp"]:
-        for entry in root.rglob(ext):
-            entry.unlink()
-
-else:
-    if cythonize is None:
-        raise ImportError("Cython is required to build from source")
+def get_gdal_paths():
+    """Obtain the paths for compiling and linking with the GDAL C-API
+
+    First the presence of the GDAL_INCLUDE_PATH and GDAL_LIBRARY_PATH environment
+    variables is checked. If they are both present, these are taken.
+
+    If one of the two paths was not present, gdal-config is called (it should be on the
+    PATH variable). gdal-config provides all the paths.
+
+    If no environment variables were specified or gdal-config was not found,
+    no additional paths are provided to the extension. It is still possible
+    to compile in this case using custom arguments to setup.py.
+    """
+    include_dir = os.environ.get("GDAL_INCLUDE_PATH")
+    library_dir = os.environ.get("GDAL_LIBRARY_PATH")
+
+    if include_dir and library_dir:
+        return {
+            "include_dirs": [include_dir],
+            "library_dirs": [library_dir],
+            "libraries": ["gdal_i" if platform.system() == "Windows" else "gdal"],
+        }
```

Comment:
So the question is whether in this case we want to require that `GDAL_VERSION` env variable is set as well? (in which case we can check that to raise an error it too old)

Or do we only want to require this linux/mac? (and on windows still try `gdalinfo --version` first)